pipeline/molecular_phenotypes/bulk_expression.ipynb

```diff
@@ -343,7 +343,7 @@
     "# Path to the software:\n",
     "parameter: soft_dir = path\n",
     "# Path to the reference adaptors:\n",
-    "parameter: adapter = str\n",
+    "parameter: adapter = \"ILLUMINACLIP:TruSeq3-PE.fa:2:30:10\"\n",
```

Comment:
Is it typically to have a default adaptor, or should we always ask people to input an adaptor? It seems particularly specific. We make something default only if it is a reasonable choice. We dont make defaults in order to lighten the interface.

Reply:
fastqc should be able to tell what adaptor is there. But this is also part of the wip that haven't implemented yet.

Reply:
fastqc is an interactive tool -- sounds like users need to run that on their own to determine the adaptor? In that case adapter should be made a required parameter without default. And we need to document how to get that sequence. It's not clear how this is obtained from the biostars link ... Do you know what to do here?

Reply:
> fastqc is an interactive tool -- sounds like users need to run that on their own to determine the adaptor? In that case adapter should be made a required parameter without default. And we need to document how to get that sequence. It's not clear how this is obtained from the biostars link ... Do you know what to do here?

one of the existed step involved running fastqc on the fq file to generates the html version of the report and a zip file containing the text description and images that are used in the report. If the adaptor information can be found by fastqc, it should be in there. But I am still figuring out what exactly to extract.

Reply:
Thank you @hsun3163 . I hope you can figure it out. I'll merge the PR after you figure it out and document it clear enough for users to locate and use it under your instructions. Let me know if you have an issue with it (we can use expertise from our other colleagues)

Reply:
OK it seems the Trimmomatic setup is more complicated than the just pasting a specific name of adaptor. 
I made a mistake here assuming `ILLUMINACLIP:TruSeq3-PE.fa:2:30:10` is a name. but in realty in means:
```
1. Cut Illumina adaptor
2. The adaptor is the more common version of Truseq3, pair-ended
3.maximum mismatch count = 2
4.  The accuracy of the match between the two 'adapter ligated' reads must be for PE palindrome read alignment = 30.
5.  The accuracy of the match between any adapter etc. sequence must be against a read = 10
```
which makes them 5 parameters instead of 1.


The rule to extract such info from fastqc report, based on trimmonia, is as followed:
```
As a rule of thumb newer libraries will 
use TruSeq3, but this really depends on your service provider.
If you use FASTQC, the „Overrepresented Sequences‟ report can help indicate which adapter 
file is best suited for your data. 
“Illumina Single End” or “Illumina Paired End” sequences indicate single-end or paired-end 
TruSeq2 libraries, and the appropriate adapter files are “TruSeq2-SE.fa” and “TruSeq2-PE.fa” 
respectively. 
“TruSeq Universal Adapter” or “TruSeq Adapter, Index …” indicates TruSeq-3 libraries, and 
the appropriate adapter files are “TruSeq3-SE.fa” or “TruSeq3-PE.fa”, for single-end and 
paired-end data respectively. 
Adapter sequences for TruSeq2 multiplexed libraries, indicated by “Illumina Multiplexing 
…”, and the various RNA library preparations are not currently included.
```
One thing to that catch my attention is `the various RNA library preparations are not currently included`, so I am not sure as in what adapter option to use

Reply:
There is the scenario where none of the aforementioned tags are reported. For example, in our provided dataset, following are the `Overrepresented Sequences` results, no hit means none of the known Truseq adaptors are there. Which may hints toward the  `the various RNA library preparations are not currently included` case.

```
>>Overrepresented sequences     fail
#Sequence       Count   Percentage      Possible Source
NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN     158     1.58    No Hit
GCTCAGGCTGGAGTGCAGTGGCTATTCACAGGCGCGATCCCACTACTGAT      29      0.29    No Hit
CTCAGGCTGGAGTGCAGTGGCTATTCACAGGCGCGATCCCACTACTGATC      25      0.25    No Hit
GCCGGGTTGAATCCTCCGGGCGGACTGCGCGGACCCCACCCGTTTACCTC      25      0.25    No Hit
GCCCTCTTGAACTCTCTCTTCAAAGTTCTTTTCAACTTTCCCTTACGGTA      22      0.22    No Hit
CCAGGCTGGAGTGCAGTGGCTATTCACAGGCGCGATCCCACTACTGATCA      19      0.19    No Hit
CAGGCTGGAGTGCAGTGGCTATTCACAGGCGCGATCCCACTACTGATCAG      18      0.18    No Hit
GGCTGGAGTGCAGTGGCTATTCACAGGCGCGATCCCACTACTGATCAGCA      17      0.16999999999999998     No Hit
GTCGGGTCTGCGAGAGCGCCAGCTATCCTGAGGGAAACTTCGGAGGGAAC      16      0.16    No Hit
CCCCTCCTTAGGCAACCTGGTGGTCCCCCGCTCCCGGGAGGTCACCATAT      13      0.13    No Hit
CCCTCCTTAGGCAACCTGGTGGTCCCCCGCTCCCGGGAGGTCACCATATT      12      0.12    No Hit
CCTTAGGCAACCTGGTGGTCCCCCGCTCCCGGGAGGTCACCATATTGATG      12      0.12    No Hit
CCCTCTTGAACTCTCTCTTCAAAGTTCTTTTCAACTTTCCCTTACGGTAC      12      0.12    No Hit
GGTTCTTCGAATGTGTGGTAGGGTGGGGGGCATCCATATAGTCACTCCAG      11      0.11    No Hit
```

Reply:
@hsun3163 thank you for the detailed investigation on this. Very enlightening. So indeed we should make all of them parameters, taken from http://www.usadellab.org/cms/?page=trimmomatic

```
Step options:

ILLUMINACLIP:<fastaWithAdaptersEtc>:<seed mismatches>:<palindrome clip threshold>:<simple clip threshold>
fastaWithAdaptersEtc: specifies the path to a fasta file containing all the adapters, PCR sequences etc. The naming of the various sequences within this file determines how they are used. See below.
seedMismatches: specifies the maximum mismatch count which will still allow a full match to be performed
palindromeClipThreshold: specifies how accurate the match between the two 'adapter ligated' reads must be for PE palindrome read alignment.
simpleClipThreshold: specifies how accurate the match between any adapter etc. sequence must be against a read.
SLIDINGWINDOW:<windowSize>:<requiredQuality>
windowSize: specifies the number of bases to average across
requiredQuality: specifies the average quality required.
LEADING:<quality>
quality: Specifies the minimum quality required to keep a base.
TRAILING:<quality>
quality: Specifies the minimum quality required to keep a base.
CROP:<length>
length: The number of bases to keep, from the start of the read.
HEADCROP:<length>
length: The number of bases to remove from the start of the read.
MINLEN:<length>
length: Specifies the minimum length of reads to be kept.
```

However only make `fastaWithAdaptersEtc` required parameter, and  tell people that this is a file name and it can be generated using fastqc (?). Other parameters should still be explicit parameters but we put in reasonable default values. The user interface will then require `--fastaWithAdaptersEtc` and nothing else.

Reply:
The problem is, as outlined in the example, sometimes we cant tell what the adapter is by fastqc though.

Reply:
Sure. And if that is the case we should automatically skip the trim. @floutt I will pin you tomorrow on the other repo to ask about the data.